services/ec2/drivers/Ec2ElbCommon.py

```diff
@@ -65,4 +66,31 @@ def _checkWAFEnabled(self):
         if 'WebACL' not in results:
             self.results['ELBEnableWAF'] = [-1, 'Disabled']
         
+        return
+    
+    def _checkALBSGPortMatch(self):
+        ## NLB not supported
+        if self.elb['Type'] != 'application':
+            return
+        
+        arn = self.elb['LoadBalancerArn']
+        results = self.elbClient.describe_listeners(
+            LoadBalancerArn = arn
+        )
+        
+        portList = []
+        for listener in results.get('Listeners'):
+            portList.append(listener.get('Port'))
+            
+        unmatchPortList = portList
+        
+        flaggedSGs = []
+        for group in self.sgList:
+            for perm in group.get('IpPermissions'):
+                if perm.get('FromPort') != perm.get('ToPort') or perm.get('FromPort') not in portList:
+                    flaggedSGs.append(group.get('GroupId'))
+        
+        if len(flaggedSGs) > 0:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`

[Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)